pipeline-scripts/release.groovy

```diff
@@ -452,6 +456,39 @@ def void sendReleaseCompleteMessage(Map release, int advisoryNumber, String advi
     }
 }
 
+def createAdvisoriesFor(ocpVersion) {
+    build(
+        job: "/build/advisories",
+        propagate: false,
+        parameters: [
+            string(name: "VERSION", value: ocpVersion),
+            string(name: "DATE",    value: determineNextReleaseDate(ocpVersion)),
+            string(name: "IMPETUS", value: "standard"),
+        ]
+    )
+}
+
+def determineNextReleaseDate(ocpVersion) {
+    def dayOfWeek
+    switch(ocpVersion) {
+        case "4.5":
+            dayOfWeek = DayOfWeek.MONDAY
+            break
+        case "4.4":
+            dayOfWeek = DayOfWeek.TUESDAY
+            break
+        default:
+            dayOfWeek = DayOfWeek.WEDNESDAY
+            break
+    }
+    return (
+        LocalDate.now()
+        .with(TemporalAdjusters.next(dayOfWeek))
```

Comment:
This is maybe sensitive to when the job runs. I think this can be made more robust to first go to `next(DayOfWeek.SUNDAY)` before taking `next(dayOfWeek)` to make the time the job runs more invariant (assuming we do not work on weekends :laughing: ).

Reply:
hm, that's a good point. buildvm's `/etc/localtime` is `UTC` ... It could break if we promote in too early in Beijing's morning (before 0800).

Reply:
I like the `DayOfWeek.SUNDAY` suggestion! A quick simulation, pretending we are promoting a 4.4 release today (Wed Aug 5). We'd expect the next 4.4 z-stream release date to be Tue Aug 18.
```
groovy:000> import java.time.DayOfWeek
===> java.time.DayOfWeek
groovy:000> import java.time.LocalDate
===> java.time.DayOfWeek, java.time.LocalDate
groovy:000> import java.time.format.DateTimeFormatter
===> java.time.DayOfWeek, java.time.LocalDate, java.time.format.DateTimeFormatter
groovy:000> import java.time.temporal.TemporalAdjusters
===> java.time.DayOfWeek, java.time.LocalDate, java.time.format.DateTimeFormatter, java.time.temporal.TemporalAdjusters
groovy:000> today = LocalDate.now()
===> 2020-08-05
groovy:000> dayOfWeek = DayOfWeek.TUESDAY
===> TUESDAY
groovy:000> (
groovy:001> today
groovy:002> .with(TemporalAdjusters.next(DayOfWeek.SUNDAY))
groovy:003> .with(TemporalAdjusters.next(dayOfWeek))
groovy:004> .with(TemporalAdjusters.next(dayOfWeek))
groovy:005> .format(DateTimeFormatter.ofPattern("yyyy-MMM-dd"))
groovy:006> )
===> 2020-Aug-18
groovy:000>
```

Reply:
done: https://github.com/openshift/aos-cd-jobs/pull/2307/commits/44645f885bfe4c8f75a603fa57b438e063791d26